kibana.md

```diff
@@ -1101,3 +1101,201 @@ an admin, who has just configured access for a new user. They would like
 to know if the rule(s) are configured correctly. And here comes the impersonation feature. The admin can impersonate given user in Kibana and see what the user would see if they logged in themselves. 
 
 ROR plugins support impersonation and provide UI for configuring cluster before using it. Visit the [impersonation details page](./details/impersonation.md) to know more.
+
+## Custom middleware
+### Why do we need it
+Sometimes, there is a need to handle non-standard behavior in the plugin.
+We want to make it more flexible and give the option to customize the plugin behavior by the enterprise license users to adjust the product to the business needs.
+There are two options to declare the custom middleware:
+- JS file: `readonlyrest_kbn.custom_middleware_inject_file: '/path/to/your/file.js'`
```

Comment:
Can we clarify the path can also be relative. And relative to what directory (IIRC it's the same dir where kibana.yml is found)

Reply:
As I checked it's relative to the kibana root folder, in the case of our local-testing, it's `/kibana` folder. Added the info ✔️